api/v1alpha1/hosted_controlplane.go

```diff
@@ -68,6 +68,11 @@ type HostedControlPlaneSpec struct {
 	// Etcd contains metadata about the etcd cluster the hypershift managed Openshift control plane components
 	// use to store data.
 	Etcd EtcdSpec `json:"etcd"`
+
+	// Configuration embeds resources that correspond to the openshift configuration API:
+	// https://docs.openshift.com/container-platform/4.7/rest_api/config_apis/config-apis-index.html
+	// +kubebuilder:validation:Optional
+	Configuration ClusterConfiguration `json:"configs,omitempty"`
```

Comment:
I believe by kube convention optional fields are recommended to be pointers. Though in practice you only need a pointer if you need to diff between nil and the zero value which is not the case here. Thoughts?

Reply:
Yeah, didn't think that it would make much of a difference if it was a pointe. However, I'm noticing that the json field name needs to be fixed. Will update.